### PR TITLE
Bug 1547746 - Render trailhead cards in a new portal / on the top of the page

### PR DESCRIPTION
**Pull request by rlr**

### CODE REVIEW

File: content-src/asrouter/asrouter-content.jsx [hunk 1]
```diff
@@ -319,6 +321,7 @@ export class ASRouterUISurface extends React.PureComponent {
     const {message, bundle} = this.state;
     if (!message.id && !bundle.template) { return null; }
     const shouldRenderBelowSearch = TEMPLATES_BELOW_SEARCH.includes(message.template);
+    const shouldRenderInHeader = TEMPLATES_ABOVE_PAGE.includes(message.template);
```
Comment (Mardak) [inline content-src/asrouter/asrouter-content.jsx]: Do we want to support showing the below-search snippet for about:welcome? I suppose generally, would we need to support showing multiple templates potentially in above page, below search, and in footer?

Comment (Mardak) [reply content-src/asrouter/asrouter-content.jsx]: aaron says about:welcome has its own special "snippet" anyway, so only one at a time. So I believe this approach should be okay where if trailhead is active, then nothing else

File: content-src/asrouter/asrouter-content.jsx [hunk 2]
```diff
@@ -333,7 +336,7 @@ export class ASRouterUISurface extends React.PureComponent {
           {this.renderOnboarding()}
           {this.renderSnippets()}
         </>,
-        this.portalContainer
+        shouldRenderInHeader ? this.headerPortal : this.footerPortal
```
Comment (k88hudson) [inline content-src/asrouter/asrouter-content.jsx]: Is this going to work with having to render 2 separate things at once though? i.e. onboarding on top, then cards?

Comment (k88hudson) [reply content-src/asrouter/asrouter-content.jsx]: Like don't we need  2 portals at once?

Comment (rlr) [reply content-src/asrouter/asrouter-content.jsx]: I was thinking we would render everything from `<Trailhead/>` and the CSS would take care of making the modal vs rendering inline on top of page. What you think?

Comment (rlr) [reply content-src/asrouter/asrouter-content.jsx]: So you would have `Trailhead` render something like:

```
<>
  <ModalOverlayWrapper>.....</ModalOverlayWrapper>
  <OnboardingCardsHere />
</>
```

**Pull request merged**